pkg/controllers/workapplier/availability_tracker.go

```diff
@@ -262,25 +264,28 @@ func trackServiceExportAvailability(curObj *unstructured.Unstructured) (Manifest
 	}
 
 	// Check if ServiceExport is valid and up to date
+	svcExportObj := klog.KObj(curObj)
 	validCond := meta.FindStatusCondition(svcExport.Status.Conditions, string(fleetnetworkingv1alpha1.ServiceExportValid))
-	if validCond == nil || svcExport.Generation != validCond.ObservedGeneration ||
-		meta.IsStatusConditionFalse(svcExport.Status.Conditions, string(fleetnetworkingv1alpha1.ServiceExportValid)) {
-		klog.V(2).InfoS("Still need to wait for ServiceExport to be available", "serviceExport", klog.KObj(curObj))
+	if !condition.IsConditionStatusTrue(validCond, svcExport.Generation) {
+		klog.V(2).InfoS("Still need to wait for ServiceExport to be valid", "serviceExport", svcExportObj, "validCondition", validCond)
 		return ManifestProcessingAvailabilityResultTypeNotYetAvailable, nil
 	}
 
-	weight := svcExport.Annotations[objectmeta.ServiceExportAnnotationWeight]
-	if weight != "0" {
+	weight, err := objectmeta.ExtractWeightFromServiceExport(&svcExport)
```

Comment:
could you please add a comment here to indicate why we check the annotation here?

eg,
"Updating the annotation won't change the object generation, so the current status is not reliable and we validate the annotation again here."